### ISSUE TITLE
Windows Terminal Preview is not recognized by Files

### ISSUE THREAD

**Issue by brukberhane** [files-community/Files#2009]
**Describe the bug**
Basically, Files doesn't recognize WT Preview. You may try to manyally add it to the terminal.json file but it gets removed on startup. Even changing the title from "Windows Terminal" to "Windows Terminal Preview" still makes Files remove that JSON object all together.

**To Reproduce**
1. Go to Settings > Preferences
2. Edit the file to add Windows Terminal Preview
3. Go back to Settings > Preferences after a restart of the app
4. See that the entry you added is deleted and not there.

**Expected behavior**
The expected behavior is that my entry for Windows Terminal Preview would continue to be there after a restart.

**Desktop:**
 - OS Version: Windows 10 1909 18363.1082
 - App version: v0.15.4.0

**Additional context**
 The workaround I have found for this is to change the command for the Windows Terminal JSON object from `wt.exe` to just `wt`.
EDIT:
For Clarity, this bug is only present when **ONLY** Windows Terminal Preview is installed and not also the Stable version.

**Comment by yaichenbaum** [comment]
Windows Terminal Preview is not intended to be detected by Files, have you tried using the sample profile for Windows Terminal Preview?

**Comment by brukberhane** [comment]
No I haven't tried using the sample profile for Windows Terminal Preview.. I actually can't see it on the wiki...

**Comment by yaichenbaum** [comment]
@brukberhane The profile for Windows Terminal Preview is the same Windows Terminal profile. See this [doc ](https://docs.microsoft.com/en-us/windows/terminal/command-line-arguments?tabs=windows)for more information.

**Comment by brukberhane** [comment]
@yair100 I am aware that they share the same CLI yet the option to select windows terminal instead of say powershell on files doesn't pop up and when you add teh profile to terminal.json manually, it deletes it on startup. Except when I did the workaroudn I mentioned

**Comment by yaichenbaum** [comment]
You shouldn't need to change the profile, if you change the `wt` argument in the Windows 10 settings app to point to the Preview then Files will automatically launch the Preview version.

**Comment by brukberhane** [comment]
I had changed the wt argument to point to the Preview... but it just wouldn't show up in the list of terminals

**Comment by yaichenbaum** [comment]
The list of terminals won't change, the apps basically runs the `wt` command to launch the app connected to that argument.

**Comment by brukberhane** [comment]
![image](https://user-images.githubusercontent.com/20716815/94811826-8dc4e600-03fe-11eb-984b-ddded0e64902.png)
![image](https://user-images.githubusercontent.com/20716815/94811843-91586d00-03fe-11eb-991d-ec7d0041d72e.png)

**Comment by yaichenbaum** [comment]
That looks correct, what happens when you try launching Terminal from the right click context menu?

**Comment by brukberhane** [comment]
@yair100 
![image](https://user-images.githubusercontent.com/20716815/94812093-edbb8c80-03fe-11eb-8bf0-990b20802397.png)

**Comment by brukberhane** [comment]
But when I place a new profile in there called "Windows Terminal Preview" and then replace the path `wt.exe` with just `wt`. It doesn't get deleted on startup
![image](https://user-images.githubusercontent.com/20716815/94812284-31ae9180-03ff-11eb-805a-a04739d8a23c.png)

**Comment by yaichenbaum** [comment]
Can you try adding this profile?
```
{
      "name": "Windows Terminal",
      "path": "wt.exe",
      "arguments": "-d .",
      "icon": ""
 }
```

**Comment by brukberhane** [comment]
Before restarting files after adding your config:
![image](https://user-images.githubusercontent.com/20716815/94812468-75a19680-03ff-11eb-9309-e11fcc93bf5d.png)
After:
![image](https://user-images.githubusercontent.com/20716815/94812457-70444c00-03ff-11eb-93c3-2a1cf0c2f725.png)

At this point, I feel that it should be worth mentioning that I **ONLY** have Windows Terminal Preview Installed.

**Comment by Atlasim1** [comment]
try and not using preview and the official version

**Comment by brukberhane** [comment]
@Atlasim1 Is it possilble to install the official version alongside the preview?

**Comment by Don-Vito** [comment]
@brukberhane - if you are asking if you can install both Terminal and Terminal Preview on the same PC, then the answer is absolutely yet. These are two different packages and appear separately both in Windows Store and chocolatey.

**Comment by brukberhane** [comment]
@Don-Vito I'm sorry if I didn't put this succinctly. What I mean is that if **ONLY** Windows Terminal Preview is installed, then Files doesn't automatically add the "Windows Terminal" option into the "open with terminal" options. Even if I were to manually add it into the .json file, My entry gets removed.

**Comment by Don-Vito** [comment]
> @Don-Vito I'm sorry if I didn't put this succinctly. What I mean is that if **ONLY** Windows Terminal Preview is installed, then Files doesn't automatically add the "Windows Terminal" option into the "open with terminal" options. Even if I were to manually add it into the .json file, My entry gets removed.

@brukberhane - actually you explained it quite well in the first time :blush: There is an issue that definitely requires a fix (I guess I understand the root cause of it but not sure how to fix it yet).

I tried to answer your last question, about installing preview and non-preview alongside. Probably I misread it.
But if such temporary workaround applicable in your case, you can install both Terminal and Terminal Preview on your system but use non-preview with Files.

**Comment by yaichenbaum** [comment]
@brukberhane This issue should be resolved with #2764.
